src/fsharp/FSharp.Core/control.fs

```diff
@@ -1462,6 +1462,50 @@ namespace Microsoft.FSharp.Control
                             |> unfake);
                     FakeUnit))
 
+        static member Choice(computations : Async<'T option> seq) : Async<'T option> =
+            unprotectedPrimitive(fun args ->
+                let result =
+                    try Choice1Of2 <| Seq.toArray computations
```

Comment:
Please just use a forward pipe.  We don't use much back piping in the F# compiler or FSharp.Core,